gammapy/modeling/models/spatial.py

```diff
@@ -1303,7 +1303,7 @@ def __init__(
                 # Normalize in each energy bin
                 data_sum = map.data.sum(axis=(1, 2)).reshape((-1, 1, 1))
 
-            data = map.data / data_sum
+            data = map.data / data_sum if data_sum.any() > 0 else map.data * 0.0
```

Comment:
not strong opinion but maybe we should raise a warning in the `else` situation?

Reply:
I agree. It is not clear what purpose an empty template can have, so it would be good to warn users that something is probably wrong with the input map

Reply:
With the safemask, we have always energy bin with no data. Should we warn systematically the users? Because at this end, it is expected in most case...

In fact, I saw this warning in one CI output. And in this case, there were no data at all... This fix allows to eliminate this case, but not all. And this does not change any result as Python takes care of the division per zero

Reply:
this won't work if `data_sum` is zero in one energy bin but not in the other, the fix should be :

`data = np.divide(map.data, data_sum, out=np.zeros_like(map.data), where=data_sum!=0)`